avogadro/qtplugins/plotbands/plotbandsdialog.cpp

```diff
@@ -0,0 +1,2109 @@
+/*
+ * Copyright 2007 Sandia Corporation.
+ * Under the terms of Contract DE-AC04-94AL85000, there is a non-exclusive
+ * license for use of this work by or on behalf of the
+ * U.S. Government. Redistribution and use in source and binary forms, with
+ * or without modification, are permitted provided that this Notice and any
+ * statement of authorship are reproduced on all copies.
+ */
+
+
+#include "ui_plotbandsdialog.h"
+#include "plotbandsdialog.h"
+
+#include <vtkDataObjectToTable.h>
+#include <vtkElevationFilter.h>
+#include <vtkPolyDataMapper.h>
+#include <vtkQtTableView.h>
+#include <vtkRenderer.h>
+#include <vtkRenderWindow.h>
+#include <vtkVectorText.h>
+#include <vtkPen.h>
+#include </home/carlos/Qt-Projects/Avogadro2OShared/build-openchemistry-Desktop-Default/prefix/include/vtk-6.3/vtkContextView.h>
+//#include <vtkContextView.h>
+#include <vtkWindowToImageFilter.h>
+#include <vtkPNGWriter.h>
+#include <vtkJPEGWriter.h>
+#include <vtkRenderWindowInteractor.h>
+#include <vtkRendererCollection.h>
+#include <vtkSmartPointer.h>
+#include <vtkActor.h>
+#include <vtkDoubleArray.h>
+#include <vtkStringArray.h>
+#include <vtkTextProperty.h>
+#include <vtkArray.h>
+
+#include "vtkAxis.h"
+#include "vtkBrush.h"
+#include "vtkCharArray.h"
+#include "vtkChartXY.h"
+#include "vtkAxis.h"
+#include "vtkContextScene.h"
+#include "vtkFloatArray.h"
+#include "vtkNew.h"
+#include "vtkPlotArea.h"
+#include "vtkPlot.h"
+//#include "vtkRenderWindow.h"
+#include "vtkRenderWindowInteractor.h"
+//#include "vtkSmartPointer.h"
+#include "vtkTable.h"
+
+//#include "vtkSmartPointer.h"
+
+#include <algorithm>
+
+#include <QApplication>
+#include <QWidget>
+#include <QMainWindow>
+#include <QHBoxLayout>
+#include <QDebug>
+#include <QVector>
+#include <QList>
+#include <QChar>
+//#include <QtCore/QMath>
+#include <QtMath>
+#include <QItemSelection>
+#include <QItemSelectionModel>
+#include <math.h>
+
+#include "QVTKWidget.h"
+#include "vtkQtTableView.h"
+
+#define VTK_CREATE(type, name) \
```

Comment:
This is deprecated

Reply:
Dear Marcus,

I will correct all the issues you enumerate. I apologize for the big amount of errors.

I developed these plugins to use of Avogadro2 with the Quantum Atomistics Simulations codes SIESTA and ANT (https://www.simuneatomistics.com/siesta/?view=default). The first one is very famous in the field of Solid State Physics.

Thank you very much for your time checking the errors, I reiterate my apologies.

Carlos Salgado